moped-editor/src/views/projects/projectView/ProjectComponents/ProjectComponentsList.js

```diff
@@ -1,197 +1,67 @@
-      return (
-        <ComponentListItem
-          key={component.project_component_id}
-          component={component}
-          isExpanded={isExpanded(component)}
-          onZoomClick={() => onZoomClick(component)}
-          onListItemClick={() => onListItemClick(component)}
-          Icon={
-            <ComponentIconByLineRepresentation
-              lineRepresentation={lineRepresentation}
-              color={theme.palette.primary.main}
-            />
-          }
-          selectedBorderColor={theme.palette.primary.main}
-          additionalCollapseListItems={
-            <Stack
-              spacing={2}
-              direction="row"
-              justifyContent="flex-end"
-              my={1}
-              // estimating alignment with zoom ListItemSecondaryAction button
-              mr={2.5}
-            >
-              <Tooltip title="Details">
-                <IconButton
-                  color="primary"
-                  aria-label="edit"
-                  onClick={onEditAttributes}
-                >
-                  <EditNoteOutlinedIcon />
-                </IconButton>
-              </Tooltip>
-
-              <Tooltip
-                title={
-                  isSignal
-                    ? "Signal locations can only be changed by editing the component details"
-                    : "Map"
-                }
-              >
-                {/* this span allows the tooltip to display when IconButton is disabled */}
-                <span>
-                  <IconButton
-                    color="primary"
-                    aria-label="map"
-                    onClick={onEditMap}
-                    disabled={isSignal}
-                  >
-                    <EditLocationAltOutlinedIcon
-                      color={
-                        getIsComponentMapped(component) ? undefined : "error"
-                      }
-                    />
-                  </IconButton>
-                </span>
-              </Tooltip>
-
-              <Tooltip
-                PopperProps={{
-                  disablePortal: true,
-                }}
-                open={!!copiedUrl}
-                disableFocusListener
-                disableHoverListener
-                disableTouchListener
-                title="Copied!"
-                placement="top"
-              >
-                {/* This span prevents warning about providing title prop to child of Tooltip */}
-                <span>
-                  <Tooltip title="Copy link to component" placement="bottom">
-                    <IconButton
-                      color="primary"
-                      aria-label="link"
-                      onClick={copyLinkToClipboard}
-                    >
-                      <LinkIcon />
-                    </IconButton>
-                  </Tooltip>
-                </span>
-              </Tooltip>
+    <List sx={{ paddingTop: 0 }}>
+      {allComponents.map((component) => {
```

Comment:
This change helps pave a path to virtualizing this list in the future. Previously, we handle project components and related components as two different lists which increased the odds of us seeing two separate scrollable areas. They are now combined and we define which list item component to render for each one depending on if they belong to the current project or a subproject.